### ISSUE TITLE
"on offer/complete/accept/decline/defer/fail/visit/stopover"

### ISSUE THREAD

**Issue by TheDeathStar3** [shitwolfymakes/Endless-Sky-Mission-Builder#25]
I completely understand that this project is still in beta mode, but I want to let you know and remind you that I really love this application, but one thing I do find annoying is that there is only one type of option for the trigger "on". Plus, if I upload a mission file and if it does not have an "on" and it has anything else (e.g "to"), then when I press compile the whole trigger part that is not "on" disappears because it is not recognised. If it is really easy to do then it would be nice for the file to keep all the trigger events while the other triggers have not been coded in yet.

**Comment by shitwolfymakes** [comment]
Thanks for the encouragement! I don't remember if I added support for `to`. Can you upload the log file?

**Comment by shitwolfymakes** [comment]
Double checked the ES docs and the codebase. the `to` statements fall under Conditions, and have not been implemented yet due to recursion complexity, it's on the list though. The biggest problem with that was actually getting the GUI to handle displaying and manipulating that data.

As for `on`: That is mostly complete, as you can see [here](https://github.com/shitwolfymakes/Endless-Sky-Mission-Builder/blob/master/src/model/components/Trigger.py). When I first built ESMB, the wiki documentation updates were not complete. As I move forward on the C++ Conversion, ESMB will be updated to cover any changes made.

**Comment by TheDeathStar3** [comment]
Hi, I am so sorry for not responding. I have run into another issue. I can not open the Mission file anymore. I have tried to reinstall the application but it did not work. Log:

DEBUG:root:Opening file: G:/EmptyGit/FAOFG 000.txt
DEBUG:root:	Parsing Mission file...
DEBUG:root:		MISSION FOUND: mission "FAOFG 000"
DEBUG:root:Building mission: FAOFG 000
DEBUG:root:	Mission components initializing...
DEBUG:root:		Component <class 'model.components.Deadline.Deadline'> initializing...
DEBUG:root:		Component <class 'model.components.Cargo.Cargo'> initializing...
DEBUG:root:		Component <class 'model.components.Passengers.Passengers'> initializing...
DEBUG:root:		Component <class 'model.components.Illegal.Illegal'> initializing...
DEBUG:root:		Component <class 'model.components.Repeat.Repeat'> initializing...
DEBUG:root:		Component <class 'model.components.Clearance.Clearance'> initializing...
DEBUG:root:		Component <class 'model.components.Stopover.Stopover'> initializing...
DEBUG:root:		Component <class 'model.components.Source.Source'> initializing...
DEBUG:root:		Component <class 'model.components.Destination.Destination'> initializing...
DEBUG:root:	Parsing mission...
DEBUG:root:		Parsing FAOFG 000 from file...
DEBUG:root:			Found mission display name: "Free World Council Meeting"
DEBUG:root:			Found description: Go to Bourne for a meeting with the Free Worlds Council
DEBUG:root:			Found deadline
DEBUG:root:			Found priority level
DEBUG:root:			Found where shown
